Project Pathfinder/Assets/Scripts/GenerateMaze.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+// X and Y grid positions of each maze cell
+public struct CellPosition{
+    public int _x;
+    public int _y;
+}
+
+// Data belonging to the neighbor of the current maze cell
+public struct CellNeighbor{
+    public CellPosition _position;   // Grid position of the neighbor cell
+    public WallStatus   _sharedWall; // Wall shared by the current and neighbor cells
+}
+
+// Status bits reflecting the presence of each wall within a maze cell
+[Flags]
+public enum WallStatus{
+    TOP     = 1,   // (0001) Top wall bit
+    LEFT    = 2,   // (0010) Left wall bit
+    BOTTOM  = 4,   // (0100) Bottom wall bit
+    RIGHT   = 8,   // (1000) Right wall bit
+    VISITED = 128, // (1000 0000) Cell visited status bit
+}
+
+public class GenerateMaze : MonoBehaviour
+{
+    // Create a completed randomly generated maze
+    public static WallStatus[,] Generate(int mazeWidth, int mazeHeight){
+        WallStatus[,] mazeData      = new WallStatus[mazeWidth, mazeHeight]; // Maze cell data
+        WallStatus    initialStatus = WallStatus.TOP | WallStatus.LEFT | WallStatus.BOTTOM | WallStatus.RIGHT;
+                                                                             // Initial wall statuses for every maze cell
+
+        // Initialize the wall statuses of each maze cell
+        for(int j = 0; j < mazeHeight; j++){
+            for(int i = 0; i < mazeWidth; i++){
+                mazeData[i,j] = initialStatus;
+            }
+        }
+        
+        // Generate all paths throughout the maze
+        return GeneratePaths(mazeData, mazeWidth, mazeHeight);
+    }
+
+    // Generate all paths throughout the maze
+    private static WallStatus[,] GeneratePaths(WallStatus[,] mazeData, int mazeWidth, int mazeHeight){
+        Stack<CellPosition> positionStack = new Stack<CellPosition>(); // Holds maze cell position data
+        System.Random randomNum           = new System.Random();       // Random number generator
+        CellPosition initialCell          = new CellPosition{ _x = randomNum.Next(0, mazeWidth), _y = randomNum.Next(0, mazeHeight) },
+                                                                       // Starting cell for path generation
+                     currentCell          = new CellPosition(),        // Position of the current cell being processed
+                     neighborCell         = new CellPosition();        // Position of the selected neighbor of the current cell
+        List<CellNeighbor> cellNeighbors  = new List<CellNeighbor>();  // List of unvisited neighbors to the current cell  
```

Comment:
could probably change "cellNeighbors" variable name to "unvisitedCellNeighbors" as it is a much clearer name

Reply:
Done.
changed it to "unvisitedNeighbors" so it matches the name of the function